src/pages/home/settings/DeleteAccountDialog.tsx

```diff
@@ -0,0 +1,126 @@
+import { Box, Stack, Typography } from "@mui/material";
+import { TextInputField } from "components";
+import { Button, Dialog } from "elements";
+import { Form, Formik, FormikValues } from "formik";
+import { FunctionComponent, useState } from "react";
+import * as Yup from "yup";
+import theme from "theme";
+import {
+  emptyProfile,
+  useDeleteAccountThunk,
+  useGetProfileQuery,
+} from "modules/session";
+
+const DeleteAccountDialog: FunctionComponent = () => {
+  const deleteAccountPhrase = "YES";
+  const regExpPhrase = new RegExp(`^${deleteAccountPhrase}$`);
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const { data: { id } = emptyProfile } = useGetProfileQuery();
+
+  const [deleteAccount, isLoading] = useDeleteAccountThunk();
+
+  const initialValues = {
+    confirmationStatement: "",
+  };
+
+  const validationSchema = Yup.object().shape({
+    confirmationStatement: Yup.string()
+      .required(`Must type the word "${deleteAccountPhrase}"`)
+      .matches(regExpPhrase, `Must type the word "${deleteAccountPhrase}"`),
+  });
+
+  const handleSubmit = ({ confirmationStatement }: FormikValues) => {
+    if (confirmationStatement === deleteAccountPhrase) {
+      deleteAccount({ id });
+      setIsModalOpen(false);
+    }
+  };
+
+  const handleCloseDialog = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleOpenDialog = () => {
+    setIsModalOpen(true);
+  };
+
+  return (
+    <>
+      <Button onClick={ handleOpenDialog } variant="secondary" color="magazine">
+        Delete account
+      </Button>
+      <Dialog open={ isModalOpen }>
+        <Formik
+          initialValues={ initialValues }
+          onSubmit={ handleSubmit }
+          validationSchema={ validationSchema }
+        >
+          { () => (
+            <Form>
+              <Box
+                sx={ {
+                  display: "flex",
+                  justifyContent: "center",
+                  flexDirection: "column",
+                  p: 3,
+                  rowGap: 2,
+                } }
+              >
+                <Stack sx={ { rowGap: 1 } }>
+                  <Typography variant="body2">
+                    You can still turn back...
+                  </Typography>
+                  <Typography variant="subtitle1">
+                    Once you delete your profile it&apos;s gone for good. If
+                    you&apos;re sure and would like to proceed, please confirm
+                    by typing &ldquo;YES&rdquo; below.
+                  </Typography>
+                </Stack>
+                <Stack
+                  sx={ {
+                    display: "flex",
+                    justifyContent: "start",
+                  } }
+                >
+                  <TextInputField name="confirmationStatement"></TextInputField>
```

Comment:
Not a required change but this can be self enclosing: 
`<TextInputField name="confirmationStatement" />`

Reply:
Will fix this in addition to the comment below before merge.

Reply:
I remember why I went with the extra wrapper Stack, the TextInputField uses a base HTML element and not a MUI variant. Was on a flow of keeping it within the `sx` pattern within the component. 

Will follow suit though and remove the extra Stack and use the style prop to handle this one.

Reply:
Oh I wasn't talking about the stack wrap. I was referring to this:
<TextInputField name="confirmationStatement" />

Vs

<TextInputField name="confirmationStatement"></TextInputField>

Sorry for the confusion